superset-frontend/src/explore/components/ExploreChartHeader/index.jsx

```diff
@@ -248,134 +226,142 @@ export class ExploreChartHeader extends React.PureComponent {
+                onSave={sliceUpdated}
+                slice={slice}
+              />
+            )}
+            <Tooltip id="edit-desc-tooltip" title={t('Edit chart properties')}>
+              <span
+                aria-label={t('Edit chart properties')}
+                role="button"
+                tabIndex={0}
+                className="edit-desc-icon"
+                onClick={openPropertiesModal}
               >
-                <span
-                  aria-label={t('Edit chart properties')}
-                  role="button"
-                  tabIndex={0}
-                  className="edit-desc-icon"
-                  onClick={this.openPropertiesModal}
-                >
-                  <i className="fa fa-edit" />
-                </span>
-              </Tooltip>
-              {this.props.chart.sliceFormData && (
-                <AlteredSliceTag
-                  className="altered"
-                  origFormData={this.props.chart.sliceFormData}
-                  currentFormData={formData}
-                />
-              )}
-            </StyledButtons>
-          )}
-        </div>
-        <div className="right-button-panel">
-          {chartFinished && queryResponse && (
-            <RowCountLabel
-              rowcount={Number(queryResponse.rowcount) || 0}
-              limit={Number(formData.row_limit) || 0}
-            />
-          )}
-          {chartFinished && queryResponse && queryResponse.is_cached && (
-            <CachedLabel
-              onClick={this.postChartFormData.bind(this)}
-              cachedTimestamp={queryResponse.cached_dttm}
-            />
-          )}
-          <Timer
-            startTime={chartUpdateStartTime}
-            endTime={chartUpdateEndTime}
-            isRunning={chartStatus === 'loading'}
-            status={CHART_STATUS_MAP[chartStatus]}
+                <i className="fa fa-edit" />
+              </span>
+            </Tooltip>
+            {chart.sliceFormData && (
+              <AlteredSliceTag
+                className="altered"
+                origFormData={chart.sliceFormData}
+                currentFormData={formData}
+              />
+            )}
+          </StyledButtons>
+        )}
+      </div>
+      <div className="right-button-panel">
+        {chartFinished && queryResponse && (
+          <RowCountLabel
+            rowcount={Number(queryResponse.rowcount) || 0}
+            limit={Number(formData.row_limit) || 0}
           />
-          {this.canAddReports() && this.renderReportModal()}
-          <ReportModal
-            show={this.state.showingReportModal}
-            onHide={this.hideReportModal}
-            props={{
-              userId: this.props.user.userId,
-              userEmail: this.props.user.email,
-              chart: this.props.chart,
-              creationMethod: 'charts',
-            }}
+        )}
+        {chartFinished && queryResponse && queryResponse.is_cached && (
```

Comment:
```suggestion
        {chartFinished && queryResponse?.is_cached && (
```
This can be cleaned with [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining).